src/ixdat/techniques/ec_ms.py

```diff
@@ -426,6 +429,156 @@ def siq_ecms_calibration_curve(
+            tspan (list): Timespan for which the partial current is returned.
+            tspan_bg (list): Timespan that corresponds to the background signal.
+            snr (int): signal-to-noise ratio used for Wiener deconvolution.
+        """
+        # TODO: comments in this method so someone can tell what's going on!
+
+        # grab the calibrated data
+        t_sig, v_sig = self.grab_flux(mol, tspan=tspan, tspan_bg=tspan_bg)
+        # calculate the impulse response
+        signal_response = impulse_response.model_impulse_response_from_params(
+            dt=t_sig[1] - t_sig[0], duration=t_sig[-1] - t_sig[0]
+        )
+        # this seems quite inefficient, to re-calculate the impulse response every time?
+        # TODO: store this somehow
+        # make sure signal_response and v_sig are same length for the steps below by
+        # adding zeros to the end of whichever array is shorter
+        if len(v_sig) >= len(signal_response[1]):
+            kernel = np.hstack(
+                (signal_response[1], np.zeros(len(v_sig) - len(signal_response[1])))
+            )
+            v_sig_corr = v_sig
+        else:
+            kernel = signal_response[1]
+            v_sig_corr = np.hstack(
+                (v_sig, np.zeros(len(signal_response[1]) - len(v_sig)))
+            )
+        H = fft(kernel)
+        # TODO: store this as well.
+        partial_current = np.real(
+            ifft(fft(v_sig_corr) * np.conj(H) / (H * np.conj(H) + (1 / snr) ** 2))
+        )
+        partial_current = partial_current * sum(kernel)  # what does this do????
+        if len(t_sig) < len(partial_current):
+            delta = len(partial_current) - len(t_sig)
+            partial_current = partial_current[:-delta]
+        return t_sig, partial_current
+
+    def extract_impulse_response(self, mol, tspan=None, tspan_bg=None):
+        """Extracts an ECMSImpulseResponse object from a measurement using the
+        algorithm developed by Krempl et al.
+        https://pubs.acs.org/doi/abs/10.1021/acs.analchem.1c00110
+
+        # TODO: add some option of plotting the impulse response together with the data
+        # TODO: re-add a possibility to choose cut-off potentials rather
+        than having to specify the tspan to make the existance of this method more
+        meaningful, right now it just returns an array of the calibrated signal of mol
+        normalized to the peak area
+        Args:
+            mol (str): Molecule from which the impulse response is to be extracted.
+            tspan(list): Timespan over which which the impulse response is
+                extracted.
+            tspan_bg (list): Timespan that corresponds to the background signal.
+        """
+        # grab the calibrated data
+        impulse_response = ECMSImpulseResponse(
+            mol=mol, data=self
+        ).calc_impulse_response_from_data(tspan=tspan, tspan_bg=tspan_bg)
+        return impulse_response
+
+    def deconvolute_for_tspans(
```

Comment:
This should also be a method of the `Calculator` (ie, the `ECMSImpulseResponse`.)

Reply:
leaving this unresolved since it's a not for you rather than me